File: CommandEventExample/PHP/MyApp/Events/Fooed.php
<?php

declare(strict_types=1);


namespace MyApp\Events;

use MyLibrary\Event\Base;
use MyLibrary\IPayload;

final class Fooed extends Base
{
    public function __construct(IPayload $payload)
    {
        parent::__construct($payload);
    }
}

File: CommandEventExample/CSharp/MyApp/Events/Fooed.cs
namespace MyApp.Events
{
    using System;
    
    public sealed class Fooed : MyLibrary.Event.Base<Payloads.Foo>
    {
        public Fooed(Payloads.Foo payload) : base(payload)
        {}
    }
    
    public sealed class FooedHandler : MyLibrary.Event.Handler<Fooed>
    {
        public override void Handle(Fooed evnt)
        {
            Console.WriteLine($"Fooed event fired with payload: \"{@evnt.Payload.Value}\"");
        }
    }
}